### ISSUE TITLE
Landscape fails while Prospector runs fine

### ISSUE THREAD

**Issue by wldhx** [landscapeio/landscape-issues#233]
The repository in question: https://github.com/graddfril/desktop.grabber-aggregator-refimpl

[Landscape](https://landscape.io/github/graddfril/desktop.grabber-aggregator-refimpl) dies with `Failure reason: Prospector execution failed` while local execution without any additional options (automatically picking up `.landscape.yml`) works fine.

What could I be doing wrong?

**Comment by carlio** [comment]
@wldhx I figured this out eventually - there are a combination of several bugs here! 

First, you have a syntax error here: https://github.com/graddfril/desktop.grabber-aggregator-refimpl/blob/master/graddfril/desktop/grabber_aggregator_refimpl/main.py#L8   . You have turned on vulture which normally doesn't run, and vulture crashes on that. That shouldn't happen and Landscape should at least manage to carry on if it does!

Secondly, how did you install prospector? Did you install `prospector[with_everything]`? The current version 0.11.7 has a bug where if a tool is not installed, it just runs anyway and doesn't warn. That's fixed on the master branch, coming in the next release!

So basically, this is Landscape failing to handle a tool crashing. I'll fix that; if you fix your syntax error your builds should start working again though.

**Comment by wldhx** [comment]
> syntax error

Hhm, I don't see one. Tried with REPL, it seems to be okay with it too (perhaps you've been meaning an older version (you link points to master, not the commit)?):

``` sh
$ python -c '"""Interactive part / CLI."""
import argparse
import getpass
import confuse
from graddfril.desktop.grabber_aggregator_refimpl.matrix import MatrixClient


def update_config(config: confuse.Configuration):
    """Update the config file."""
    open(config.user_config_path(), 'w').write(config.dump())
'

$
```

> how did you install prospector?
> The current version 0.11.7 has a bug where if a tool is not installed, it just runs anyway and doesn't warnsmiley: 

No, I've installed `prospector[with_vulture,with_pyroma]`. I _have_ forgotten the extras at one point, but no output at all was illegitimate enough for me to fiddle around and get was wrong :) And now I've even fixed [the requirements](https://github.com/graddfril/desktop.grabber-aggregator-refimpl/pull/17).

**Comment by carlio** [comment]
@wldhx Sorry I didn't explain properly - I meant that prospector/astroid throws a Syntax Error on that line. I think astroid doesn't handle type annotations properly yet, will look into it further!

**Comment by flother** [comment]
I'm having a similar problem (see #239) and I think it has to do with enabling Vulture. Landscape worked on [my repo](https://landscape.io/github/flother/htmltab) until I added a `vulture` block to my `.landscape.yml` file, after which it fails every time even though I've since disabled Vulture. Prospector works locally.
